Sources/RInAppMessaging/Utilities/ResponseStateMachine.swift

```diff
@@ -0,0 +1,23 @@
+import Foundation
+
+internal struct ResponseStateMachine {
+    internal enum State {
+        case success
+        case error(Error)
+    }
```

Comment:
Looks like a custom Result :)

Reply:
Hmm....true...

Reply:
@malayli  Do you think I should refactor it somehow to use the `Result` type. I don't have any good idea

Reply:
@imaginaris I don't think so except if you can set a value in case of success.

Reply:
@malayli Actually I noticed I don't need an associated value for `error`